lib/ansible/cli/galaxy.py

```diff
@@ -1083,6 +1083,7 @@ def execute_init(self):
                 company='your company (optional)',
                 license='license (GPL-2.0-or-later, MIT, etc)',
                 role_name=obj_name,
+                namespace=obj_name.split('.')[0],
```

Comment:
Actually I am a bit unclear on what a role name should be: 

* namespace.role_name
* role_name

`ansible-lint` outputs this:

> ERROR    Computed fully qualified role name of testdir does not follow current galaxy requirements.

And just refers to using the format `namespace.rolename instead. But I struggle to find that information even in the [Ansible Galaxy documentation](https://docs.ansible.com/ansible/latest/galaxy/user_guide.html). The command `ansible-galaxy` certainly takes both formats and creates the role structure in the file system.

My gutt-feeling tells me that this is either unspecified or left this flexible on purpose and probably even is a different discussion than this one.

> You can supplement/override these variables in 2.17 by using --extra-vars

I tried `ansible-galaxy role init --init-path /tmp testdir --extra-vars namespace=spreadcat` on 2.17.4, but namespace didn't show up in `meta/main.yml`. Did I misunderstand you or am I using it wrong?

Reply:
You probably need to search the ansible-lint documentation, instead of the Galaxy user guide. ansible-lint is more opinionated than the actual feature.

> I tried ansible-galaxy role init --init-path /tmp testdir --extra-vars namespace=spreadcat on 2.17.4, but namespace didn't show up in meta/main.yml. Did I misunderstand you or am I using it wrong?

You would have to provide a role skeleton https://docs.ansible.com/ansible/latest/reference_appendices/config.html#galaxy-role-skeleton that actually uses the variables you pass in. A role skeleton looks like a role, except any .j2 files outside of the templates/ dir will be templated during `ansible-galaxy role init`. Correct usage could look like `ansible-galaxy role init --role-skeleton /path/to/skeleton --extra-vars @/path/to/vars_file.yml rolename`